coalib/settings/ConfigurationGathering.py

```diff
@@ -448,6 +450,40 @@ def get_filtered_bears(languages,
     return local_bears, global_bears
 
 
+def get_style_config(args, config_file):
+    """
+    Get the style guide configuration files for all
+    the requested styles and collate them into the
+    specified config file.
+    :param args: Alternative pre-parsed CLI arguments.
+    :param config_file: The config file to collate to
+    """
+    if args and args.generate_config:
+        style_list = args.generate_config
+        base_url = ('https://raw.githubusercontent.com/PrajwalM2212/'
+                    'coala-styles/master/coala_styles/styles/')
+        file_list = []
+        from coalib.core.Bear import Bear
+        for style in style_list:
+            lang = style[0]
+            author = style[1]
+            try:
+                file_list.append(Bear.download_cached_file(
```

Comment:
Quite hacky, this should be dropped imo. It could interfere with bear caches. It is fine if we don't have caching mechanism built in as of yet. You can build a simple caching system if you so desire rn.

Also you can just work with the file contents rather than using file objects directly.